api.js

```diff
@@ -56,309 +60,310 @@ Api.prototype._runFile = function (file, runStatus, execArgv) {
 	});
 
 	var emitter = fork(file, options, execArgv);
-
 	runStatus.observeFork(emitter);
 
 	return emitter;
 };
 
+Api.prototype.run = function (files, options) {
+	var self = this;
+
+	var findOptions = {
+		cwd: this.options.resolveTestsFrom,
+		files: files
+	};
+
+	return new AvaFiles(findOptions)
+		.findTestFiles()
+		.then(function (files) {
+			return self._run(files, options);
+		});
+};
+
 Api.prototype._onTimeout = function (runStatus) {
 	var timeout = ms(this.options.timeout);
-	var message = 'Exited because no new tests completed within the last ' + timeout + 'ms of inactivity';
-
-	runStatus.handleExceptions({
-		exception: new AvaError(message),
-		file: undefined
-	});
+	var err = new AvaError('Exited because no new tests completed within the last ' + timeout + 'ms of inactivity');
+	this._handleError(runStatus, err);
 
 	runStatus.emit('timeout');
 };
 
-Api.prototype.run = function (files, options) {
+Api.prototype._setupTimeout = function (runStatus) {
 	var self = this;
+	var timeout = ms(this.options.timeout);
 
-	return new AvaFiles({files: files, cwd: this.options.resolveTestsFrom})
-		.findTestFiles()
-		.then(function (files) {
-			return self._run(files, options);
+	runStatus._restartTimer = debounce(function () {
+		self._onTimeout(runStatus);
+	}, timeout);
+
+	runStatus._restartTimer();
+	runStatus.on('test', runStatus._restartTimer);
+};
+
+Api.prototype._cancelTimeout = function (runStatus) {
+	runStatus._restartTimer.cancel();
+};
+
+Api.prototype._setupPrecompiler = function (files) {
+	var isCacheEnabled = this.options.cacheEnabled !== false;
+	var cacheDir = uniqueTempDir();
+
+	if (isCacheEnabled) {
+		cacheDir = findCacheDir({
+			name: 'ava',
+			files: files
 		});
+	}
+
+	this.options.cacheDir = cacheDir;
+
+	var isPowerAssertEnabled = this.options.powerAssert !== false;
+	this.precompiler = new CachingPrecompiler(cacheDir, this.options.babelConfig, isPowerAssertEnabled);
 };
 
-Api.prototype._run = function (files, _options) {
-	var self = this;
+Api.prototype._run = function (files, options) {
+	if (!options) {
+		options = {};
+	}
+
+	var prefixTitles = this.options.explicitTitles || files.length > 1;
+	var basePath = path.relative('.', commonPathPrefix(files)) + path.sep;
+
 	var runStatus = new RunStatus({
-		prefixTitles: this.options.explicitTitles || files.length > 1,
-		runOnlyExclusive: _options && _options.runOnlyExclusive,
-		base: path.relative('.', commonPathPrefix(files)) + path.sep
+		runOnlyExclusive: options.runOnlyExclusive,
+		prefixTitles: prefixTitles,
+		base: basePath
```

Comment:
Yes, it'd rename it to `basePath`, but in a separate PR dedicated to cleaning up `RunStatus`.